### DISCUSSION TITLE
Voice guidance broken after attaching to road

### DISCUSSION THREAD

**Discussion by lmarceg** [osmandapp/OsmAnd#16688] [Maps Related]
Hi
I normally use gpx.studio to prepare my maps and then I let OsmAnd import them. It's all perfectly working but of course voice guidance is too "talkative" this way. I recently found out the new "attach to road" function and I used it last week to test it.
Unfortunately, what I got was a correct map with correct in-map indications, but voice guidance was kind of offset by, I don't know, minutes or kilometers. So in the end it was telling me to turn right where there was no turn, or to go left instead of right. Basically, I had to use the visual map all the time. I closed and opened the app, stopped and restarted routing, but nothing happened. As soon as I stopped using "attach to road", it all went back to normal.
I tested another circuit yesterday and apart from the maybe 2 or 3 indications, which were correct, then offset started again.

I am attaching some screenshots I took while I was on my bike.
Is there something I can do to improve the situation? Because right now it's basically impossible to use this feature.

<img src="https://user-images.githubusercontent.com/1092640/224833038-cc99ec11-f05c-4f1a-b763-07db69d6777e.PNG" width="25%">  <img src="https://user-images.githubusercontent.com/1092640/224833038-cc99ec11-f05c-4f1a-b763-07db69d6777e.PNG" width="25%">  <img src="https://user-images.githubusercontent.com/1092640/224833049-9e2d8d91-8dd5-4b6f-bc73-91f58a5e9ece.PNG" width="25%">
<img src="https://user-images.githubusercontent.com/1092640/224833052-06ef9cf0-5c61-4620-891e-bb59e63b8be0.PNG" width="25%">  <img src="https://user-images.githubusercontent.com/1092640/224833058-da1578a1-6292-4b13-9aea-0ea650a69334.PNG" width="25%">  <img src="https://user-images.githubusercontent.com/1092640/224833060-d6a8e426-8653-41bd-9a4d-3ba93849dd84.PNG" width="25%">

**Comment by lmarceg** [comment]
Despite I didn't see any comments for the above, I see the attach on road was improved, because now I do not see the issues I described in March. However, still it's not working: I tested it yesterday and I observed two problems:

1) Routes are not fully attached and the new routing sometimes take an arbitrary path, as you can see for the image below where the red GPX track is completely gone unseen (note that without attaching on road, it all works well)
2) When deviating from the path, the GPX track is basically forgotten and the best path to home is calculated instead. I went off track by 3km while listening to the indications, before realizing I was going elsewhere...

<img src="https://github.com/osmandapp/OsmAnd/assets/1092640/65370f6f-510d-4d6e-951a-a4fc0153395e" width=50%>

**Comment by K-Krs** [comment]
Hi, I raised an issue on this: #17090. The issue seems to be observered, unfortunately there hasn't been assigned any milestone to solve the issue. For me, removing the waypoints from your gpx file seems to improves the navigation guidance (screen and voice). You could help the developpers of Osmand if you could tell us whether the gpx file in both your examples includes waypoint or not.
